File: NET/RestSharpTests/RestSharpTests/testsuite_sample/ValidateTextInSms.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Nodes;
using System.Threading.Tasks;
using AventStack.ExtentReports;
using RestSharp;
using RestSharpTests;
using RestSharpTests.Base;
using RestSharpTests.Utilties;
using Newtonsoft.Json;
using AventStack.ExtentReports.Model;
using NUnit.Framework.Interfaces;
using Newtonsoft.Json.Linq;
using NUnit.Framework.Internal;

namespace RestSharpTests.testsuite_sample
{
    [TestFixture]
    public class ValidateTextInSms : BaseClass
    {
        [Test]
        [TestCaseSource(typeof(TestDataProvider), nameof(TestDataProvider.GetTestData), new object[] { "Test1" })]
        public void SendSmsInEnglish(string dataSetAsString)
        {
            try
            {
               Console.WriteLine(dataSetAsString);
                JObject dataMap = JObject.Parse(dataSetAsString);
                restRequest = restSharpUtility.GetRestRequest(ObjectHelper.SendSampleApiEndpoint);
                restRequest = restSharpUtility.addBody(restRequest, dataMap["requestBody"].ToString());
                restRequest = restSharpUtility.addQueryParam(restRequest, "phoneNumber", dataMap["queryParams"]["phoneNumber"].ToString());
                //get response

                restResponse = restSharpUtility.GetRestResponse(restRequest, Method.Post);
                Assert.IsTrue(Validation.ResponseCode(extentTest, (int)restResponse.StatusCode, 200));


                Assert.IsTrue(Validation.ResponseBody(extentTest, restResponse.Content, dataMap["expectedResponse"].ToString()));

              
            }
            catch (Exception e)
            {
                LogMessage(Status.Fail, "Test failed :" + e.ToString());
                Assert.IsTrue(false, e.ToString());

            }
        }
    }
}

File: untitled/src/test/java/testSuite_Sample/ValidateTextInSms.java
package testSuite_Sample;

import com.aventstack.extentreports.Status;
import io.restassured.http.Headers;
import io.restassured.response.Response;
import org.example.BaseClass;
import org.example.ObjectHelper;
import org.example.TestDataProvider;
import org.example.Validation;
import org.json.JSONObject;
import org.testng.Assert;
import org.testng.ITestResult;
import org.testng.Reporter;
import org.testng.annotations.Test;

public class ValidateTextInSms extends BaseClass {

    @Test(dataProvider = "testData", dataProviderClass = TestDataProvider.class,groups = {"Test1"})
    public void SendSmsInEnglish(String dataSetAsString){
        try {
            JSONObject dataMap = new JSONObject(dataSetAsString);
            //test.log(Status.INFO, dataMap.getString("testDescripion");
            System.out.println(dataMap.getString("testDescription"));

            //Request builder and response
            ReqSpecs= RaUtils.getReqSpecs();
            ReqSpecs = RaUtils.addBasePath(ReqSpecs, ObjectHelper.sendSampleApiEndpoint);
            ReqSpecs = RaUtils.addStringBodyToReq(ReqSpecs,dataMap.get("requestBody").toString());
            ReqSpecs= RaUtils.addPathParam(ReqSpecs,"phoneNumber",dataMap.getJSONObject("queryParams").getString("phoneNumber"));
            //get response

            Response resp = RaUtils.getResponse(ReqSpecs,"post");
            Assert.assertTrue(Validation.responseCode(test, resp.getStatusCode(), 200));

            //getting response headers
        //    Headers responseHeaders = resp.getHeaders();
         //   Assert.assertTrue(Validation.headers(test,responseHeaders,ObjectHelper.allHeaders));

            Assert.assertTrue(Validation.responseBody(test,resp.getBody().asString(), String.valueOf(dataMap.get("expectedResponse"))));

        }catch (Exception e){
            Reporter.getCurrentTestResult().setStatus(ITestResult.FAILURE);
            test.log(Status.FAIL,"Test failed :" + e.toString());
            Assert.assertTrue(false,e.toString());
        }
    }
}
//https://api.apis.guru/v2/providers.json